src/client/dfs/dfs.c

```diff
@@ -1206,11 +1239,11 @@ open_sb(daos_handle_t coh, bool create, daos_obj_id_t super_oid,
 	}
 
 	if (iods[2].iod_size != sizeof(layout_ver) ||
-	    layout_ver != DFS_LAYOUT_VERSION) {
+	    layout_ver > DFS_LAYOUT_VERSION) {
 		D_ERROR("Incompatible DFS Layout version.\n");
 		D_GOTO(err, rc = EINVAL);
 	}
-
+	*ver = layout_ver;
```

Comment:
I think this will segfault if called from dfs_cont_create_int()

Reply:
it shouldn't get here in that case as it's called with create option that returns above.

Reply:
That explains why it passes testing then.  If course it wouldn't get here, if it's creating a container then it wouldn't need to fetch the version anyway.